### PR TITLE
Node garbage collection

### PR DESCRIPTION
**Pull request by evanxg852000**
Implemented dead nodes garbage collection.

https://github.com/quickwit-oss/scuttlebutt/issues/17

### CODE REVIEW

File: scuttlebutt/src/state.rs [hunk 1]
```diff
@@ -130,6 +130,10 @@ impl ClusterState {
         self.seed_nodes.iter().map(|node_id| node_id.as_str())
     }
 
+    pub fn remove_node(&mut self, node_id: &NodeId) {
```
Comment (fulmicoton) [inline scuttlebutt/src/state.rs:133]: should that be pub?

File: scuttlebutt/src/failure_detector.rs [hunk 2]
```diff
@@ -72,7 +72,7 @@ impl FailureDetector {
             debug!(node_id = ?node_id, phi = phi, "updating node liveliness");
             if phi > self.config.phi_threshold {
                 self.live_nodes.remove(node_id);
```
Comment (fulmicoton) [inline scuttlebutt/src/failure_detector.rs:74]: super ultra nitpick.

I like the idea to emphasive on the fact taht we move the node from live to dead.

for instance via:

```
if let Some(node_id) = self.live_nodes.remove(node_id) {
    self.dead_nodes.insert(node_id, Instant::now());
}
```

Comment (fulmicoton) [reply scuttlebutt/src/failure_detector.rs:74]: ( no need to fix that'ss probably just me)

**Review by fulmicoton — APPROVED** [review]

File: scuttlebutt/src/failure_detector.rs (hunk 2 above)
Comment (evanxg852000) [reply scuttlebutt/src/failure_detector.rs:74]: This might not be a nitpick. it really makes the code speaks for itself TBH. which is always great to have

**Pull request merged**